Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/tabs/ContactDiaryDayTab.kt

```diff
@@ -0,0 +1,35 @@
+package de.rki.coronawarnapp.contactdiary.ui.day.tabs
+
+import androidx.fragment.app.Fragment
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.location.ContactDiaryLocationListFragment
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.location.ContactDiaryLocationListFragmentArgs
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.person.ContactDiaryPersonListFragment
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.person.ContactDiaryPersonListFragmentArgs
+
+data class ContactDiaryDayTab(
+    val tabNameResource: Int,
+    val fabTextResource: Int,
+    val fragmentInstantiation: (day: Long) -> Fragment
+) {
+    companion object {
+        val PERSON_TAB = ContactDiaryDayTab(
+            R.string.contact_diary_day_person_tab_title,
+            R.string.contact_diary_day_person_fab_title
+        ) { day ->
+            ContactDiaryPersonListFragment().apply {
+                // Feels kind of hacky but i like the free typesafety for the args
```

Comment:
I'm open to suggestions on how to do this differently, i kind of like the typesafety here but if anyone has a different idea on how to pass the current day to the sub fragments feel free to comment :)

Reply:
Maybe sealed classes?

```kotlin
sealed class DiaryTab {
val selectedDay...
}
```

and the viewmodel of the screen that hosts the view pager offers an observable that generates these `tab` objects.

Otherwise the instantiation like this looks fine.

Reply:
hmm not 100% sure on how i would pass the day to the subfragment then

Reply:
Sealed class has no impact on instantiation you could still instantiate it like you do now, just removes the static instances.